iocore/net/BIO_fastopen.cc

```diff
@@ -58,26 +57,26 @@ fastopen_bwrite(BIO *bio, const char *in, int insz)
 
   errno = 0;
   BIO_clear_retry_flags(bio);
-  ink_assert(bio->num != NO_FD);
+  ink_assert(BIO_get_fd(bio, nullptr) != NO_FD);
```

Comment:
Rather than calling `BIO_get_fd` multiple times, call it once and stash the fd in a local variable. Same for the write path.